### PR TITLE
smtp-configuration form validation added.

### PR DESCRIPTION
**Pull request by velmuruganvelayutham**
Added form validation while saving SMTP configuration parameters.

### CODE REVIEW

**Comment by syjer** [comment]
@velmuruganvelayutham thanks for the PR :), tomorrow I'll have a look

File: src/main/webapp/partials/admin/manage-smtp-configuration.html [hunk 1]
```diff
@@ -11,50 +11,71 @@
 			</div>
 		</div>
 		<div class="body separated">
+		<form name="smtpForm" data-ng-submit="smtpForm.$valid && saveSmtpConfig(configuration)" novalidate>
 			<div class="form-group">
 				<div class="lvg-form-label" data-translate>partials.admin.manage-smtp-configuration.host</div> 
-				<input class="form-control" type="text" data-ng-model="configuration.host" required>
+				<input class="form-control" type="text" name="host" data-ng-model="configuration.host" required>
+				 <div class="has-error" data-ng-show="smtpForm.$submitted || smtpForm.host.$touched">
+                      <span class="help-block" data-ng-show="smtpForm.host.$error.required">host is required !.</span>
```
Comment (syjer) [inline src/main/webapp/partials/admin/manage-smtp-configuration.html]: "host is required !" should be moved in messages_en.properties

File: src/main/webapp/partials/admin/manage-smtp-configuration.html [hunk 2]
```diff
@@ -11,50 +11,71 @@
...
+                 </div>
 			</div>
+			  
 			<div class="form-group">
 				<div class="lvg-form-label" data-translate>partials.admin.manage-smtp-configuration.port</div> 
-				<input class="form-control" type="number" data-ng-model="configuration.port" required>
+				<input class="form-control" type="number" name="port" data-ng-model="configuration.port" required>
+				<div class="has-error" data-ng-show="smtpForm.$submitted || smtpForm.port.$touched">
+                      <span class="help-block" data-ng-show="smtpForm.port.$error.required">port is required !.</span>
```
Comment (syjer) [inline src/main/webapp/partials/admin/manage-smtp-configuration.html]: hardcoded text "port is required !." should be moved in messages_en.properties.

File: src/main/webapp/partials/admin/manage-smtp-configuration.html [hunk 3]
```diff
@@ -11,50 +11,71 @@
...
 			</div>
 			<div class="form-group">
 				<div class="lvg-form-label" data-translate>partials.admin.manage-smtp-configuration.protocol</div> 
-				<select class="form-control" type="text" data-ng-model="configuration.protocol" required>
+				<select class="form-control" type="text" name="protocol" data-ng-model="configuration.protocol" required>
 					<option value="smtp">smtp</option>
 					<option value="smtps">smtps</option>
 				</select>
+				<div class="has-error" data-ng-show="smtpForm.$submitted || smtpForm.protocol.$touched">
+                      <span class="help-block" data-ng-show="smtpForm.protocol.$error.required">protocol is required !.</span>
```
Comment (syjer) [inline src/main/webapp/partials/admin/manage-smtp-configuration.html]: hardcoded text should be moved in messages_en.properties.

File: src/main/webapp/partials/admin/manage-smtp-configuration.html [hunk 4]
```diff
@@ -11,50 +11,71 @@
...
+                 </div>
 			</div>
 			<div class="form-group">
 				<div class="lvg-form-label" data-translate>partials.admin.manage-smtp-configuration.username</div> 
-				<input class="form-control" type="text" data-ng-model="configuration.username">
+				<input class="form-control" type="text" data-ng-model="configuration.username" name="username" required>
+				<div class="has-error" data-ng-show="smtpForm.$submitted || smtpForm.username.$touched">
+                      <span class="help-block" data-ng-show="smtpForm.username.$error.required">username is required !.</span>
```
Comment (syjer) [inline src/main/webapp/partials/admin/manage-smtp-configuration.html]: hardcoded text should be moved in messages_en.properties.

File: src/main/webapp/partials/admin/manage-smtp-configuration.html [hunk 5]
```diff
@@ -11,50 +11,71 @@
...
+                 </div>
 			</div>
 			<div class="form-group">
 				<div class="lvg-form-label" data-translate>partials.admin.manage-smtp-configuration.password</div>
-				<input class="form-control" type="password" data-ng-model="configuration.password">
+				<input class="form-control" type="password" data-ng-model="configuration.password" name="password" required>
+				<div class="has-error" data-ng-show="smtpForm.$submitted || smtpForm.password.$touched">
+                      <span class="help-block" data-ng-show="smtpForm.password.$error.required">password is required !.</span>
```
Comment (syjer) [inline src/main/webapp/partials/admin/manage-smtp-configuration.html]: hardcoded text should be moved in messages_en.properties.

File: src/main/webapp/partials/admin/manage-smtp-configuration.html [hunk 6]
```diff
@@ -11,50 +11,71 @@
...
+                 </div>
 			</div>
 	
 			<div class="form-group">
 				<div class="lvg-form-label" data-translate>partials.admin.manage-smtp-configuration.from</div>
-				<input class="form-control" type="email" data-ng-model="configuration.from" required>
+				<input class="form-control" type="email" name="from" data-ng-model="configuration.from" required>
+				<div class="has-error" data-ng-show="smtpForm.$submitted || smtpForm.from.$touched">
+                      <span class="help-block" data-ng-show="smtpForm.from.$error.required">from E-Mail is required !.</span>
```
Comment (syjer) [inline src/main/webapp/partials/admin/manage-smtp-configuration.html]: hardcoded text should be moved in messages_en.properties.

**Comment by syjer** [comment]
@velmuruganvelayutham , could you move the hardcoded text (see the comment in the review) in messages_en.properties files?

You should put the keys near "partials.admin.manage-smtp-configuration.host" :).

After you have updated the PR, I'll merge it :)

Thank you.

**Comment by velmuruganvelayutham** [comment]
@syjer I agree with all your review comments !. Done the changes and pushed it..

**Comment by syjer** [comment]
@velmuruganvelayutham thank you :)

**Pull request merged**